File: python/rna-transcription/rna_transcription.py
RNA_MAP = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
}

def to_rna(dna_strand):
    return str.join('', map(lambda k: RNA_MAP[k], list(dna_strand)))

File: typescript/rna-transcription/rna-transcription.ts
export const MAPPING: { [index: string]: string } = {
  A: "U",
  T: "A",
  C: "G",
  G: "C",
};

export function toRna(input: string) {
  const inputList = [...input];
  const keys = Object.keys(MAPPING);
  const valid = inputList.every((x) => {
    return keys.includes(x);
  });
  if (valid) {
    return inputList.map((x) => MAPPING[x]).join("");
  } else {
    throw new Error("Invalid input DNA.");
  }
}
